File: src/main/java/com/gitlab/daring/image/command/feature/ORBCommand.java
package com.gitlab.daring.image.command.feature;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import com.gitlab.daring.image.command.parameter.DoubleParam;
import com.gitlab.daring.image.command.parameter.IntParam;
import org.bytedeco.javacpp.opencv_features2d.ORB;

class ORBCommand extends BaseCommand {

    //TODO support all params
    final IntParam maxFeatures = intParam(100, "0-100");
    final DoubleParam scale = doubleParam(12, "0-100");
    final IntParam nlevels = intParam(8, "0-100");
    final ORB detector = ORB.create();

    ORBCommand(String... args) {
        super(args);
    }

    @Override
    public void execute(CommandEnv env) {
        detector.setMaxFeatures(maxFeatures.v);
        detector.setScaleFactor(scale.v * 0.1);
        detector.setNLevels(nlevels.v);
        env.featureDetector = detector;
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/feature/ORBCommand.kt
package com.gitlab.daring.image.command.feature

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import org.bytedeco.javacpp.opencv_features2d.ORB

internal class ORBCommand(vararg args: String) : KBaseCommand(*args) {

    //TODO support all params
    val maxFeatures = intParam(100, "0-100")
    val scale = doubleParam(12.0, "0-100")
    val nlevels = intParam(8, "0-100")
    val detector = ORB.create()

    override fun execute(env: CommandEnv) {
        detector.maxFeatures = maxFeatures.v
        detector.scaleFactor = scale.v * 0.1
        detector.nLevels = nlevels.v
        env.featureDetector = detector
    }

}